pkg/flowcli/gateway/grpc.go

```diff
@@ -61,37 +65,35 @@ func (g *GrpcGateway) GetAccount(address flow.Address) (*flow.Account, error) {
 	return account, nil
 }
 
-// TODO: replace with txsender - much nicer implemented
-// SendTransaction send a transaction to flow
-func (g *GrpcGateway) SendTransaction(tx *flow.Transaction, signer *project.Account) (*flow.Transaction, error) {
-	account, err := g.GetAccount(signer.Address())
+// PrepareTransactionPayload prepares the payload for the transaction from the network
+func (g *GrpcGateway) PrepareTransactionPayload(tx *project.Transaction) (*project.Transaction, error) {
```

Comment:
I'm not sure if I think it makes sense for `PrepareTransactionPayload` to be implemented in the gateway layer. It contains core logic that shouldn't need to change between gateways.

Reply:
That is true, so should I just fetch the data in the service layer and assign it to the transaction model? I put it here cuz I didn't want to bloat the service layer but yes it makes sense to avoid code duplication.